autoscale/src/main/java/com/sequenceiq/periscope/repository/ScalingActivityRepository.java

```diff
@@ -18,32 +19,39 @@
 
 @EntityType(entityClass = ScalingActivity.class)
 @Transactional(Transactional.TxType.REQUIRED)
-public interface ScalingActivityRepository extends CrudRepository<ScalingActivity, Long> {
+public interface ScalingActivityRepository extends PagingAndSortingRepository<ScalingActivity, Long> {
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.activityCrn = :activityCrn")
     Optional<ScalingActivity> findByActivityCrn(@Param("activityCrn") String activityCrn);
 
-    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId")
-    List<ScalingActivity> findAllByCluster(@Param("clusterId") Long clusterId);
+    @Query("SELECT st FROM ScalingActivity st WHERE (st.cluster.stackName = :clusterNameOrCrn OR st.cluster.stackCrn = :clusterNameOrCrn)")
+    List<ScalingActivity> findAllByCluster(@Param("clusterNameOrCrn") String clusterNameOrCrn, Pageable pageable);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.startTime < :startTimeBefore ORDER BY st.startTime DESC")
     List<ScalingActivity> findAllByClusterWithStartTimeBefore(@Param("clusterId") Long clusterId, @Param("startTimeBefore") Date startTimeBefore);
 
-    @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.startTime >= :startTimeAfter ORDER BY st.startTime DESC")
-    List<ScalingActivity> findAllByClusterWithStartTimeAfter(@Param("clusterId") Long clusterId, @Param("startTimeAfter") Date startTimeAfter);
+    @Query("SELECT st FROM ScalingActivity st WHERE (st.cluster.stackName = :clusterNameOrCrn OR st.cluster.stackCrn = :clusterNameOrCrn)" +
+            " AND st.startTime >= :startTimeAfter ORDER BY st.startTime DESC")
+    List<ScalingActivity> findAllByClusterWithStartTimeAfter(@Param("clusterNameOrCrn") String clusterNameOrCrn,
+            @Param("startTimeAfter") Date startTimeAfter, Pageable pageable);
 
     @Query("SELECT st FROM ScalingActivity st WHERE st.cluster.id = :clusterId AND st.activityStatus = :activityStatus AND st.startTime >= :startTimeFrom " +
             "AND st.startTime < :startTimeUntil ORDER BY st.startTime DESC")
     List<ScalingActivity> findAllByClusterAndActivityStatusBetweenInterval(@Param("clusterId") Long clusterId,
             @Param("activityStatus") ActivityStatus activityStatus, @Param("startTimeFrom") Date startTimeFrom, @Param("startTimeUntil") Date startTimeUntil);
 
+    @Query("SELECT st FROM ScalingActivity st WHERE (st.cluster.stackName = :clusterNameOrCrn OR st.cluster.stackCrn = :clusterNameOrCrn)" +
```

Comment:
Please split into separate name and crn queries. See `AutoScaleClusterCommonService#getClusterByCrnOrName` https://github.com/hortonworks/cloudbreak/blob/e16ff1c96f9339a58644bf7cac029b7db9195621/autoscale/src/main/java/com/sequenceiq/periscope/controller/AutoScaleClusterCommonService.java#L199-L205

Reply:
Done